blockchain/genesis_test.go

```diff
@@ -26,38 +26,124 @@ import (
 	"testing"
 
 	"github.com/davecgh/go-spew/spew"
+	"github.com/klaytn/klaytn/blockchain/vm"
 	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/consensus/gxhash"
 	"github.com/klaytn/klaytn/params"
 	"github.com/klaytn/klaytn/storage/database"
+	"github.com/stretchr/testify/assert"
 )
 
-// TODO-Klaytn-FailedTest Failed test
-/*
 func TestDefaultGenesisBlock(t *testing.T) {
-	block := DefaultGenesisBlock().ToBlock(nil)
-	if block.Hash() != params.MainnetGenesisHash {
-		t.Errorf("wrong mainnet genesis hash, got %v, want %v", block.Hash(), params.MainnetGenesisHash)
+	block := genCypressGenesisBlock().ToBlock(common.Hash{}, nil)
+	if block.Hash() != params.CypressGenesisHash {
+		t.Errorf("wrong cypress genesis hash, got %v, want %v", block.Hash(), params.CypressGenesisHash)
 	}
-	block = DefaultBaobabGenesisBlock().ToBlock(nil)
-	if block.Hash() != params.TestnetGenesisHash {
-		t.Errorf("wrong testnet genesis hash, got %v, want %v", block.Hash(), params.TestnetGenesisHash)
+	block = genBaobabGenesisBlock().ToBlock(common.Hash{}, nil)
+	if block.Hash() != params.BaobabGenesisHash {
+		t.Errorf("wrong baobab genesis hash, got %v, want %v", block.Hash(), params.BaobabGenesisHash)
 	}
 }
-*/
 
-func TestSetupGenesis(t *testing.T) {
-	var (
-		// TODO-Klaytn-FailedTest Uncomment after enabling failed tests
-		// customghash = common.HexToHash("0x89c99d90b79719238d2645c7642f2c9295246e80775b38cfd162b696817fbd50")
-		customg = Genesis{
-			Config: &params.ChainConfig{},
-			Alloc: GenesisAlloc{
-				{1}: {Balance: big.NewInt(1), Storage: map[common.Hash]common.Hash{{1}: {1}}},
+func TestHardCodedChainConfigUpdate(t *testing.T) {
```

Comment:
If you may describe what this test is for, it would be better to understand.

Reply:
Added comments at https://github.com/klaytn/klaytn/pull/1017/commits/93f41f7a96224bbb924a3420e3579ab1c040f50c.